File: src/main/java/net/atos/zac/app/taken/model/RESTTaakVerdelenGegevens.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.taken.model;

import java.util.List;

public class RESTTaakVerdelenGegevens {

  public List<RESTTaakVerdelenTaak> taken;

  public String behandelaarGebruikersnaam;

  public String groepId;

  public String reden;
}

File: src/main/kotlin/net/atos/zac/app/taken/model/RESTTaakVerdelenGegevens.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.taken.model

import jakarta.validation.constraints.NotBlank
import nl.lifely.zac.util.AllOpen
import nl.lifely.zac.util.NoArgConstructor

@AllOpen
@NoArgConstructor
data class RESTTaakVerdelenGegevens(
    var taken: List<RESTTaakVerdelenTaak>,

    @field:NotBlank
    var groepId: String,

    var behandelaarGebruikersnaam: String? = null,

    var reden: String? = null,

    /**
     * Unique screen event resource ID which can be used
     * to track the progress of the 'assign taken from list' asynchronous job
     * using web sockets.
     */
    var screenEventResourceId: String? = null
)
